scripts/jenkins/ardana/gerrit/gerrit_merge.py

```diff
@@ -24,6 +24,43 @@ def check_all_dependencies_satisfied(change):
 
 
 def gerrit_merge(change):
+    """
+    Attempt to merge a Gerrit change.
+
+    The following criteria must be met first for a Gerrit change to
+    be merged:
+
+    1. the 'status' value must be NEW (not MERGED and ABANDONED, obviously)
+    2. the 'mergeable' flag must be set, meaning there are no merge conflicts
+    between the Gerrit change and the target branch.
+    3. the 'submittable' flag must be set, meaning that the change has been
+    approved by the project submit rules when:
+
+      - at least one Code-Review+2 label value
+      - no Code-Review-2 label values
+      - at least one Verified+2 label value
+      - no Verified-2 label value
+      - at least one Workflow+1 label value
+      - no Workflow-1 label value
+
+    NOTE: if this change is submitted together with other changes, the
+    'submittable' and 'mergeable' flags will not keep track of those changes.
+    They only reflect the status of the current change.
```

Comment:
I think we are protected by the checks in check_all_dependencies_satisfied(), but if a change is submitted with unmerged parent commits, will they be submitted along with it and therefore merged? If so, does gerrit check they are submittable?

Reply:
Yes, if a change B is uploaded together and on top of A, submitting B will not be possible until A is submittable (has the necessary number of votes, no conflicts, etc.) and when the submit button is pushed for B, A will also be merged. This is where the "submitted together" notion comes from.

Reply:
Thanks, good to know.